pilot/pkg/bootstrap/server.go

```diff
@@ -184,10 +221,26 @@ func NewServer(args PilotArgs) (*Server, error) {
 	if err := s.initClusterRegistries(&args); err != nil {
 		return nil, fmt.Errorf("cluster registries: %v", err)
 	}
+
+	if err := s.initDNSListener(&args); err != nil {
+		return nil, fmt.Errorf("istiod: %v", err)
+	}
+
+	// Will run the sidecar injector in pilot.
+	// Only operates if /var/lib/istio/inject exists
 	if err := s.initSidecarInjector(&args); err != nil {
 		return nil, fmt.Errorf("sidecar injector: %v", err)
 	}
 
+	if err := s.initSDSCA(&args); err != nil {
```

Comment:
also missing flag guard

Reply:
This is actually guarded by the presence of the JWT token - but I can add an explicit flag as well.

In other words: Istiod expects to have the K8S JWT mounted, just like workloads. If Pilot doesn't have the token, because K8S doesn't have such support - the SDS feature will be off, and we need to either not use mTLS or rely on a CA that creates secrets ( like Citadel ).